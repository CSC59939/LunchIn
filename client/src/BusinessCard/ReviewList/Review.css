.review-card{
  display: flex;
  width: 100%;
  height: 100%;
}
.review-card-media{
  object-fit: cover;
  width: 100%;
  height: 100%;
}
.review-content{
  padding: 1%;
  overflow: auto;
}
