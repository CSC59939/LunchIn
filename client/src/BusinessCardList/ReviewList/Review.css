.review-card{

}
.review-card-media{
  width: 200px;
  height: 150px;
}
