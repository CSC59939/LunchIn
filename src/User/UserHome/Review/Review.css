.review-card{
  max-width: 345px,
}
.review-card-media{
  width: 200px;
  height: 150px;
}
