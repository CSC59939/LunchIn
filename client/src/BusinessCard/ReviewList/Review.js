import React, { Component } from "react";
import Card from '@material-ui/core/Card';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import "./Review.css";

class Review extends Component {
  render() {
    if(this.props.review){
      const r = this.props.review;
      return (
        <Card className="review-card">
            <Grid container>
              <Grid item xs={4} sm={4} md={4} lg={4} xl={4}>
                <img
                  className="review-card-media"
                  src={r.user.image_url}
                  alt="Failed to load review URL."
                />
              </Grid>
              <Grid item xs={8} sm={8} md={8} lg={8} xl={8}
                className="review-content"
                >
                <Typography variant="h6" component="h6">
                  {r.user.name}
                </Typography>
                <Typography component="p">
                  {r.time_created.split(' ')[0]} Rating: {r.rating} <br />
                  {r.text}
                </Typography>
              </Grid>
            </Grid>
        </Card>
      );
    }
    else{
      return (<CircularProgress />);
    }
  }
}
export default Review;
