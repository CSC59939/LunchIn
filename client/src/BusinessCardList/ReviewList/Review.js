import React, { Component } from "react";
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import "./Review.css";

class Review extends Component {
  render() {
    if(this.props.review){
      return (
        <Card className="review-card">
            <CardMedia
              className="review-card-media"
              image={this.props.review.user.image_url}
              title={this.props.review.user.image_url}
            />
            <Typography component="p">
              {this.props.review.user.name} <br />
              {this.props.review.time_created} <br />
              {this.props.review.rating} <br />
              {this.props.review.text} <br />
            </Typography>
        </Card>
      );
    }
    else{
      return (<CircularProgress />);
    }
  }
}
export default Review;
