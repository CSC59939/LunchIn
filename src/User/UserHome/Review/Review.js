import React, { Component } from "react";
import "./Review.css";

/*
  This component requires two properties: term, location
*/
class Review extends Component{
  constructor(props){
    super(props);
    this.state = {

    };
    this.renderReview = this.renderReview.bind(this);
    this.getReviews = this.getReviews.bind(this);
    this.getReviews();
  }

  getReviews(){
		fetch(`/api/yelp?term=${this.props.term}&location=${this.props.location}`)
			.then(response => response.json())
			.then(data => {
				console.log(data);
				this.setState({businesses: data.jsonBody.businesses});
			})
			.catch(e => console.log(e));
  }

  renderReview(review){
    return (
      <div className="review" key={review.id}>
        {/*console.log(review)*/}
        <div className="review-name">{review.name}</div>
        <div className="review-price">Price: {review.price}</div>
        <div className="review-rating">Rating: {review.rating}</div>
        <div className="review-review_count">Count: {review.review_count}</div>
        <div><img src={review.image_url} className="review-img"></img></div>
      </div>
    );
  }

  render(){
    return (
      <div className="review-list">
        { this.props.term
          ? (
              <div className="selected-food">
                We have selected the following food: {this.props.term}
              </div>
            )
          : (
              <div className="error">"Unable to pick a lunch..."</div>
            )
        }
        { this.state.businesses
          ? (this.state.businesses.map(this.renderReview))
          : ("")
        }
      </div>
    );
  }
}
export default Review;
