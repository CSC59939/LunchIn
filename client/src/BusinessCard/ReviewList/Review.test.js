import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Review from "./Review";
import toJson, { shallowToJson, mountToJson, renderToJson } from 'enzyme-to-json';

describe("Review Component", () => {
  describe("When provided with no info", () => {
    it("Should render correctly", () => {
      const reviewInstance = shallow(
        <Review />
      );
      expect(shallowToJson(reviewInstance)).toMatchSnapshot();
    });
  });
  describe("When provided with review info", () => {
    it("Should render correctly", () => {
      const review =
      {
        "id": "xAG4O7l-t1ubbwVAlPnDKg",
        "rating": 5,
        "user": {
          "id": "W8UK02IDdRS2GL_66fuq6w",
          "profile_url": "https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w",
          "image_url": "https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg",
          "name": "Ella A."
        },
        "text": "Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...",
        "time_created": "2016-08-29 00:41:13",
        "url": "https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w"
      };
      const reviewInstance = shallow(
        <Review review={review} />
      );
      expect(shallowToJson(reviewInstance)).toMatchSnapshot();
    });
  });
});
