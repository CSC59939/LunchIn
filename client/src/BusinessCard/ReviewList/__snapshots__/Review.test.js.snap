// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component When provided with no info Should render correctly 1`] = `<WithStyles(CircularProgress) />`;

exports[`Review Component When provided with review info Should render correctly 1`] = `
<WithStyles(Card)
  className="review-card"
>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      lg={4}
      md={4}
      sm={4}
      xl={4}
      xs={4}
    >
      <img
        alt="Failed to load URL."
        className="review-card-media"
        src="https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="review-content"
      item={true}
      lg={8}
      md={8}
      sm={8}
      xl={8}
      xs={8}
    >
      <WithStyles(Typography)
        component="h6"
        variant="h6"
      >
        Ella A.
      </WithStyles(Typography)>
      <WithStyles(Typography)
        component="p"
      >
        2016-08-29
         Rating: 
        5
         
        <br />
        Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...
      </WithStyles(Typography)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Card)>
`;
